### DISCUSSION TITLE
Do I need to memoize the `file` parameter if it is a string?

### DISCUSSION THREAD

**Discussion by mjuopperi** [wojtekmaj/react-pdf#1600] [Q&A] [answered]
The documentation for `Document` states:

> Since equality check (`===`) is used to determine if `file` object has changed, it must be memoized by setting it in component's state, `useMemo` or other similar technique.

However when using `file`, the check should work without memoizing the value in my component. Is there something I am missing here?

**Comment by wojtekmaj** [comment, marked as answer]
No, you don't need to memoize a string. This is because:

```
{} === {} // false
"foo" === "foo" // true
```

so your gut feeling was correct :D
